eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -354,12 +401,13 @@ public synchronized BackupRegistry get() {
                     1, clientConfig.getCacheRefreshExecutorThreadPoolSize(), 0, TimeUnit.SECONDS,
                     new SynchronousQueue<Runnable>());  // use direct handoff
 
-            scheduleServerEndpointTask(zone);
-
             fetchRegistryGeneration = new AtomicLong(0);
 
             clientAccept = EurekaAccept.fromString(clientConfig.getClientDataAccept());
 
+            eurekaTransport = new EurekaTransport();
+            scheduleServerEndpointTask(eurekaTransport, zone);
```

Comment:
All the new transport stuff are created lazily so will not use any resources unless triggered. We do need them to be pre-created though, as currently the option to use/no-use them are dynamic and not tied to a restart.

Having said that, the latest change to the AsyncResolver may have broken the lazy guarantee. Will fix that.